### PR TITLE
reference/performance: add a use case for SPM evolve explanation

### PR DESCRIPTION
**Pull request by eurekaka**
### What is changed, added or deleted? (Required)

Add an example to make SPM evolve easier to understand

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: reference/performance/execution-plan-bind.md [hunk 1]
```diff
@@ -98,6 +98,37 @@ set global tidb_evolve_plan_baselines = on;
 
 `tidb_evolve_plan_baselines` 的默认值为 `off`。
 
+> **注意：**
+>
+> 设置 global 系统变量在当前 session 并不会生效，只会在新创建的 session 生效。可以将 global 关键字替换为 session 使当前 session 开启演进功能。
+
 在打开自动演进功能后，如果优化器选出的最优执行计划不在之前绑定的执行计划之中，会将其记录为待验证的执行计划。每隔 `bind-info-lease`（默认值为 `3s`），会选出一个待验证的执行计划，将其和已经绑定的执行计划中代价最小的比较实际运行时间。如果待验证的运行时间更优的话，会将其标记为可使用的绑定。
 
+用一个例子描述这个过程如下：
```
Comment (TomShawn) [inline reference/performance/execution-plan-bind.md]:
```suggestion
用以下示例描述上述过程：
```

File: reference/performance/execution-plan-bind.md [hunk 2]
```diff
@@ -98,6 +98,37 @@ set global tidb_evolve_plan_baselines = on;
...
+假如有一张表定义为
```
Comment (TomShawn) [inline reference/performance/execution-plan-bind.md]:
```suggestion
假如有表 `t` 定义如下：
```

File: reference/performance/execution-plan-bind.md [hunk 3]
```diff
@@ -98,6 +98,37 @@ set global tidb_evolve_plan_baselines = on;
...
+
+{{< copyable "sql" >}}
+
+```sql
+create table t(a int, b int, key(a), key(b));
+```
+
+对于查询
```
Comment (TomShawn) [inline reference/performance/execution-plan-bind.md]:
```suggestion
在表 `t` 上进行如下查询：
```

File: reference/performance/execution-plan-bind.md [hunk 4]
```diff
@@ -98,6 +98,37 @@ set global tidb_evolve_plan_baselines = on;
...
+
+{{< copyable "sql" >}}
+
+```sql
+select * from t where a < 100 and b < 100;
+```
+
+表上满足条件 `a < 100` 的行很少，但由于某些原因，优化器没能选中使用索引 `a` 这个最优执行计划，而是误选了很慢的全表扫，那么用户首先可以通过如下语句创建一个绑定：
```
Comment (TomShawn) [inline reference/performance/execution-plan-bind.md]:
```suggestion
表上满足条件 `a < 100` 的行很少。但由于某些原因，优化器没能选中使用索引 `a` 这个最优执行计划，而是误选了速度慢的全表扫，那么用户首先可以通过如下语句创建一个绑定：
```

File: reference/performance/execution-plan-bind.md [hunk 5]
```diff
@@ -98,6 +98,37 @@ set global tidb_evolve_plan_baselines = on;
...
+
+{{< copyable "sql" >}}
+
+```sql
+create global binding for select * from t where a < 100 and b < 100 using select * from t use index(a) where a < 100 and b < 100;
+```
+
+当这条查询再次执行时，优化器会在刚创建绑定的干预下选择使用索引 `a` 进而降低查询时间。假如随着表的插入和修改，表中满足条件 `a < 100` 的行变得越来越多，而满足条件 `b < 100` 的行变得越来越少，这时再在绑定的干预下使用索引 `a` 可能就不是最优了。绑定的演进可以解决这类问题，当优化器感知到表数据变化后，会对这条查询生成使用索引 `b` 的执行计划，但由于绑定的存在，这个执行计划不会被采纳和执行，不过它会被存在后台的演进列表里。在演进过程中，如果它被验证为执行时间明显低于使用索引 `a` 的执行时间（即绑定的执行计划），那它会被加入到可用的绑定列表中。在此之后，当这条查询再次被执行时，优化器首先生成使用索引 `b` 的执行计划，并确认它在绑定列表中，所以会采纳它并执行，进而可以在数据变化后降低这条查询的执行时间。
```
Comment (TomShawn) [inline reference/performance/execution-plan-bind.md]:
```suggestion
当以上查询语句再次执行时，优化器会在刚创建绑定的干预下选择使用索引 `a`，进而降低查询时间。

假如随着在表中进行插入和修改，表中满足条件 `a < 100` 的行变得越来越多，而满足条件 `b < 100` 的行变得越来越少，这时再在绑定的干预下使用索引 `a` 可能就不是最优了。

绑定的演进可以解决这类问题。当优化器感知到表数据变化后，会对这条查询生成使用索引 `b` 的执行计划。但由于绑定的存在，这个执行计划不会被采纳和执行，不过它会被存在后台的演进列表里。在演进过程中，如果它被验证为执行时间明显低于使用索引 `a` 的执行时间（即当前绑定的执行计划），那么索引 `b` 会被加入到可用的绑定列表中。在此之后，当这条查询再次被执行时，优化器首先生成使用索引 `b` 的执行计划，并确认它在绑定列表中，所以会采纳它并执行，进而可以在数据变化后降低这条查询的执行时间。
```

**Comment by TomShawn** [comment]
@eurekaka Please address the above comments, thanks!

**Comment by ChristianLeow** [comment]
@yikeke: I'd like to translate this PR.

**Comment by yikeke** [comment]
> @yikeke: I'd like to translate this PR.

Sure, welcome! I'm assigning this PR to you.

**Pull request merged**